core/src/test/java/org/apache/spark/JavaAPISuite.java

```diff
@@ -762,6 +762,20 @@ public void min() {
   }
 
   @Test
+  public void naturalMax() {
+    JavaDoubleRDD rdd = sc.parallelizeDoubles(Arrays.asList(1.0, 2.0, 3.0, 4.0));
+    double max = rdd.max();
+    Assert.assertEquals(4.0, max, 0.001);
```

Comment:
I think you can assert exact equality here. There's no reason that it would be merely close to 4.0.